### PR TITLE
Wal improvements

### LINKED ISSUE #39: Proper logging framework
Using a proper logging framework should make debugging easier and make the output look nicer.

### LINKED ISSUE #42: DeletedTable Response
Add a DeletedTable to Response enum. As a response when a Table has been deleted.

https://github.com/algebraicdb/algebraicdb/blob/32d204074fd8c338a134560318a1dc4b25d632ae/src/local/types.rs#L36-L41

### LINKED ISSUE #81: Remove use of global AtomicU64 in wal.rs
This global variable means that having multiple instances of the dbms running in the same process will result in bugs.

### LINKED ISSUE #83: Replace channels in dbms_state and wal with locks
For our use cases, a lock is faster than a channel, and more ergonomic.

### LINKED ISSUE #84: Truncate WAL when it gets too big

### LINKED ISSUE #85: Clean up old snapshots

### PR DESCRIPTION
**Pull request by hulthe**
- [x] implement DROP TABLE
- [x] truncate WAL when it gets too big
- [x] replace channels with locks
- [ ] ~create the data dir on startup (is this even desirable?)~
- [x] validate data dir on startup
- [x] clean up old snapshots when snapshotting
- [x] clean up old/broken files on startup (.tmp-files & old snapshots)
- [x] use a proper logging framework

resolves #83 
resolves #81 
resolves #84 
resolves #39 
resolves #85 
resolves #42

### CODE REVIEW

**Comment by hulthe** [comment]
I'm going to keep working on this, but as things stand it should be in a merge-able state.
Therefore, I am once asking for your ~financial support~ code reviews.

File: algebraicdb/src/executor/dbms/mod.rs [hunk 1]
```diff
@@ -323,6 +325,18 @@ async fn execute_create_type(
     Ok(())
 }
 
+async fn execute_drop_table(
+    drop: Drop<'_>,
+    s: &DbmsState,
+    w: &mut (dyn AsyncWrite + Send + Unpin),
+) -> Result<(), Box<dyn Error>> {
+    match s.drop_table(drop.table).await {
+        Ok(_table) => w.write_all(b"Table created\n").await?,
```
Comment (Sorchar) [inline algebraicdb/src/executor/dbms/mod.rs]: I think you mean table deleted!

Comment (hulthe) [reply algebraicdb/src/executor/dbms/mod.rs]: Fixed!

**Review by Baloo1 — APPROVED** [review]

**Review by Sorchar — APPROVED** [review]
Looks good! Let me know when it's updated

**Comment by hulthe** [comment]
Aight, its time to merge, assuming the code is good.

**Review by feffes — APPROVED** [review]

**Pull request merged**